src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorThreadContainer.java

```diff
@@ -46,13 +46,18 @@ public static MonitorThreadContainer getInstance() {
     public static synchronized MonitorThreadContainer getInstance(IExecutorServiceInitializer executorServiceInitializer) {
         if (singleton == null) {
             singleton = new MonitorThreadContainer(executorServiceInitializer);
+            CLASS_USAGE.getAndSet(0);
         }
-        classUsage.getAndIncrement();
+        CLASS_USAGE.getAndIncrement();
         return singleton;
     }
 
     public static synchronized void releaseInstance() {
-        if (classUsage.decrementAndGet() <= 0) {
+        if (singleton == null) {
+            return;
+        }
+
+        if (CLASS_USAGE.decrementAndGet() <= 0) {
```

Comment:
Can this value go below 0?

Reply:
This value shouldn't be able to go below 0, but we need to make sure we cover the case for when it does go below zero.